src/Node/Internal.hs

```diff
@@ -1141,32 +1192,38 @@ spawnHandler stateVar provenance action =
                       _nodeStateOutboundBidirectional = Map.alter alteration peer (_nodeStateOutboundBidirectional nodeState)
                     }
                     where
-                    alteration Nothing = Just $ Map.singleton nonce (someHandler, channelIn, peerDataVar, connBundle, False)
-                    alteration (Just map) = Just $ Map.insert nonce (someHandler, channelIn, peerDataVar, connBundle, False) map
+                    alteration Nothing = Just $ Map.singleton nonce (someHandler, channelIn, incrBytes, peerDataVar, connBundle, False)
+                    alteration (Just map) = Just $ Map.insert nonce (someHandler, channelIn, incrBytes, peerDataVar, connBundle, False) map
                 Local peer Nothing -> nodeState {
                       _nodeStateOutboundUnidirectional = Map.insert peer someHandler (_nodeStateOutboundUnidirectional nodeState)
                     }
 
+            incrBytes !n = do
+                nodeState <- readSharedAtomic stateVar
+                stIncrBytes (handlerProvenancePeer provenance) n (_nodeStateStatistics nodeState)
+                modifySharedAtomic totalBytes $ \(!m) -> return (m + n, ())
```

Comment:
> Ah, not anymore :)
>
> https://www.stackage.org/haddock/lts-8.2/base-4.9.1.0/src/GHC-MVar.html#readMVar
>
> But it will be the same problem on older GHC versions. Can we live with that?

Nice, wasn't aware of this. It's new in 4.9.1.0? I suppose it's not such a big deal if older versions have slightly worse performance, as long as they have the same semantics, which they do.